src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Implementation/Policy/NewAzurePolicyAssignment.cs

```diff
@@ -112,11 +132,88 @@ private JToken GetResource()
                 {
                     DisplayName = this.DisplayName ?? null,
                     PolicyDefinitionId = this.PolicyDefinition.Properties["policyDefinitionId"].Value.ToString(),
-                    Scope = this.Scope
+                    Scope = this.Scope,
+                    Parameters = GetParameters()
                 }
             };
 
             return policyassignmentObject.ToJToken();
         }
+
+        object IDynamicParameters.GetDynamicParameters()
+        {
+            if (PolicyDefinition != null)
+            {
+                var properties = PolicyDefinition.Properties["Properties"].Value as PSObject;
+                if (properties != null)
+                {
+                    var parameters = properties.Properties["parameters"].Value as PSObject;
+                    if (parameters != null)
+                    {
+                        foreach (var param in parameters.Properties)
+                        {
+                            var type = (param.Value as PSObject).Properties["type"].Value.ToString();
+                            var dp = new RuntimeDefinedParameter
+                            {
+                                Name = param.Name,
+                                ParameterType = type.Equals("string", StringComparison.OrdinalIgnoreCase) ? typeof(string) : typeof(object[])
+                            };
+                            dp.Attributes.Add(new ParameterAttribute
+                            {
+                                Mandatory = false,
+                                ValueFromPipelineByPropertyName = true,
+                                HelpMessage = param.Name
+                            });
+                            dynamicParameters.Add(param.Name, dp);
+                        }
+                    }
+                }
+            }
+
+            return dynamicParameters;
+        }
+
+        protected JObject GetParameters()
+        {
+            // Load parameters from local file or literal
+            if (PolicyParameters != null)
+            {
+                string policyParameterFilePath = this.TryResolvePath(PolicyParameters);
```

Comment:
I think it is confusing to use a single parameter as maybe a file name or maybe a string representing parameters.
We should explicitly make them two separate parameters. So in total we should have three parameters:
PolicyParameters
PolicyParametersFile
PolicyParametersObject.
Each should be in its own ParameterSet so that user can only specify one of them.